### PR TITLE
feat(tui): Implement conditional option

### LINKED ISSUE #171: Implement the conditional option
Implement a conditional option where scicookie will show the question just if the dependency was defined.

so initial entry in the configuration can be found here: https://github.com/osl-incubator/scicookie/blob/main/src/scicookie/profiles/base.yaml#L213

maybe it will need a refined structured.

### PR DESCRIPTION
**Pull request by Anavelyz**
## Pull Request description

If a user chooses certain options, this may lead to a new question or new options. To do this, it was necessary to implement the conditional option in `ui.py`.

Fixes #171

## Pull Request checklists

This PR is a:

- [ ] bug-fix
- [x] new feature
- [ ] maintenance

About this PR:

- [ ] it includes tests.
- [ ] the tests are executed on CI.
- [ ] the tests generate log file(s) (path).
- [ ] pre-commit hooks were executed locally.
- [ ] this PR requires a project documentation update.

Author's checklist:

- [ ] I have reviewed the changes and it contains no misspelling.
- [ ] The code is well commented, especially in the parts that contain more
      complexity.
- [ ] New and old tests passed locally.

## Reviewer's Checklist

- [ ] I managed to reproduce the problem locally from the `main` branch
- [ ] I managed to test the new changes locally
- [ ] I confirm that the issues mentioned were fixed/resolved

### CODE REVIEW

File: src/scicookie/cookiecutter.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
   "use_ruff": "no",
   "use_shellcheck": "no",
   "use_vulture": "no",
+  "use_git": "yes",
```
Comment (xmnlab) [inline src/scicookie/cookiecutter.json:62]:
```suggestion
  "use_git": "no",
```

it should be no by default in the cookiecutter.json

**Comment by xmnlab** [comment]
It is a good progress @Anavelyz ! thanks for working on that

checking the workflow, I think that if the user decide to not configure git, it should not add anything with git.

it will be a bit challenge ... but at least I think you should review https://github.com/osl-incubator/scicookie/blob/main/src/scicookie/hooks/post_gen_project.py#L328

maybe it should abort or skip the git commands there 

maybe it should also remove the .github folder

**Comment by xmnlab** [comment]
does it make sense?

**Comment by Anavelyz** [comment]
This was an attempt to resolve issue #171, unfortunately it was not successful so I will close the PR.

**Pull request closed without merging**